src/renderer/renderElement.ts

```diff
@@ -99,25 +100,43 @@ export function renderElement(
     rc.draw(element.shape as Drawable);
     context.globalAlpha = 1;
   } else if (element.type === "arrow") {
-    const [x1, y1, x2, y2, x3, y3, x4, y4] = getArrowPoints(element);
     const options = {
       stroke: element.strokeColor,
       strokeWidth: element.strokeWidth,
       roughness: element.roughness,
       seed: element.seed,
     };
 
+    // points array can be empty in the beginning, so it is important to add
+    // initial position to it
+    const points =
+      element.points.length > 0 ? element.points : ([[0, 0]] as Point[]);
+
     if (!element.shape) {
+      const [x2, y2, x3, y3, x4, y4] = getArrowPoints(element);
+
       element.shape = [
         //    \
         generator.line(x3, y3, x2, y2, options),
-        // -----
-        generator.line(x1, y1, x2, y2, options),
+        // generate lines
+        generator.curve(points, options),
         //    /
         generator.line(x4, y4, x2, y2, options),
       ];
     }
 
+    if (element.isSelected) {
+      // draw segment endpoints when element is selected
+      points.forEach(p => {
+        rc.ellipse(p[0], p[1], 10, 10, {
+          fillStyle: "solid",
+          fill: "#ffffff",
+          strokeWidth: 2,
+          roughness: 0,
+        });
+      });
+    }
```

Comment:
As mentioned by @dwelle, let's not display them until we actually can change them. Otherwise it's confusing. Also, they are very big right now, we may want to use a different style (maybe the same square as the handles)

Reply:
Done!